src/controllers/director.residential.address.search.controller.ts

```diff
@@ -121,11 +122,19 @@ export const getCountryFromKey = (country: string): string => {
 }
 
 const renderPage = (res: Response, req: Request, officerFiling : OfficerFiling, validationErrors: ValidationError[]) => {
+  const postcode = {
+    [PostcodeKey]: req.body.postcode
+  };
+  const premises = {
+    [PremiseKey]: req.body.premises
```

Comment:
Shouldn't we be getting these values from mongo rather than the req? The state of the req may not be consistent if a user moves back and forth in the journey which will lead to entered data being lost

Reply:
Thanks Jack.
In this specific case the validation is done in frontend side rather than api side. We don't have anything in filing / mongo unless all validation is clear.
it's similar to what's done under[ src/controllers/remove.director.controller.ts](https://github.com/companieshouse/officer-filing-web/blob/main/src/controllers/remove.director.controller.ts#L126-L128).

Reply:
I suspect that will go against GDS requirements to maintain the state of the page. Might need a chat with Richard/Phil

Reply:
Thanks Jack for inputs on this. I have updated changes to patch the filing the use patched data when rendering, this will make sure if user decides to navigate after errors, the data will be present in textboxes when they navigate back to this page.